src/codegen_internal.h

```diff
@@ -1,6 +1,6 @@
 // This file is a part of Julia. License is MIT: http://julialang.org/license
 
-#if defined(LLVM38) && !defined(LLVM37)
+#if defined(JL_LLVM_VERSION) && JL_LLVM_VERSION >= 30800
```

Comment:
IMHO, we shouldn't check if `JL_LLVM_VERSION` is defined. This can catch errors of missing include earlier rather than silently ignoring the error.